compiler/codegen/src/compile.rs

```diff
@@ -1249,21 +1292,21 @@ impl Compiler {
     fn compile_function_def(
         &mut self,
         name: &str,
-        args: &located_ast::Arguments,
-        body: &[located_ast::Stmt],
-        decorator_list: &[located_ast::Expr],
-        returns: Option<&located_ast::Expr>, // TODO: use type hint somehow..
+        parameters: &Parameters,
+        body: &[Stmt],
+        decorator_list: &[Decorator],
+        returns: Option<&Expr>, // TODO: use type hint somehow..
         is_async: bool,
-        type_params: &[located_ast::TypeParam],
+        type_params: Option<&TypeParams>,
     ) -> CompileResult<()> {
         self.prepare_decorators(decorator_list)?;
 
         // If there are type params, we need to push a special symbol table just for them
-        if !type_params.is_empty() {
+        if !type_params.is_some() {
             self.push_symbol_table();
         }
```

Comment:
@qingshi163 @youknowone: This is the source of the bug.
As the comment says, `// If there are type params, we need to push a special symbol table just for them`, but the change to `!type_params.is_some()` (as in `type_params.is_none()`) pushes if there are NO params, causing the `panic!` in `push_symbol_table`. On my machine changing to
```rust 
  if type_params.is_some() {
      self.push_symbol_table();
  }
``` 
allows `test_compile_phello` to pass.